### ISSUE TITLE
Please provide a complete list of files and folders generated by VSCode and their locations

### ISSUE THREAD

**Issue by TryerGit** [microsoft/vscode#193261]
On re-opening a project and navigating to the terminal via `Ctrl + backtick` (Toggle Terminal), I notice that the terminal states:

` *  History restored `

This would mean that there is some file/folder on disk somewhere which tracks this project and stores this data. 

Can a complete list of files and folders generated by VSCode and their locations be documented and put up on the web? My hard disk space is running out and I would like to keep deleting such extra folders routinely so that it does not eat into the remaining space.

In other words, what files and folders can one safely delete so as to restore VSCode to the status it would have been under right after the very first time it has been freshly installed on a new machine without any extensions, cache files, etc. without affecting its functionality.

**Comment by Tyriar** [comment]
Where VS Code puts settings is the "user data directory" (the `Code` folder in the examples here) https://code.visualstudio.com/docs/getstarted/settings#_settings-file-locations, this is the place where everything is cached. It will probably get annoying if you delete this though as you'll get all first time notifications.

The only other place is the extensions directory which is in `~/.vscode/extensions`. Note that extensions themselves can put things anywhere, generally extensions only put things into those 2 folders or the opened folder though.

You might be interested in portable mode which is a more self contained version https://code.visualstudio.com/docs/editor/portable

**Comment by TryerGit** [comment]
@Tyriar 

In Windows under WSL, should one be looking at `$HOME/.config/Code` folder under WSL? I don't have the `Code` folder under `.config` there. Or would it be the case that working on `%APPDATA%\Code\` under the Windows directory structure suffices to delete the files created on the machine even if one had opened a project under WSL?

Thanks.

**Comment by Tyriar** [comment]
If your using the remote wsl extension it's a little more complicated, there will also be a .vscode-server (or similar) in the root with a corresponding extensions dir

The easiest way to find these directories is the open extensions folder command, and opening up the settings.json file (user or remote) and checking it's path.

**Comment by TryerGit** [comment]
@Tyriar 

I am now interested in the Portable Mode VSCode. The page you linked to only says what to do to migrate over for pure Windows and pure Linux. I, however, also work on WSL on my Windows box and notice that there is VSCode data that is written to `\\wsl$\Ubuntu-20.04\home\TryerGit\.vscode-server` How should this be migrated to the portable mode?

Is there a way to work on WSL also portably?

**Comment by Tyriar** [comment]
@TryerGit I don't think there is

**Comment by TryerGit** [comment]
> @TryerGit I don't think there is

I guessed as much. I posted this on SO and the suggestion there was to think of installing VSCode on WSL as if it were a completely new Linux only installation -- not related to the Windows install. See that thread [here](https://stackoverflow.com/questions/77176490/how-do-i-migrate-a-vs-code-installation-on-wsl-to-portable-mode)

Interesting, there was a user here before who had raised this issue a year ago (see [here](https://github.com/microsoft/vscode/issues/145854#issuecomment-1076076314)) but his solution would not work because in WSL he was trying to provide the link to the full VSCode Windows installation and not the portable installation.

This is a disappointment since I was hopeful portable VSCode would avoid the trouble of having an ever growing collection of cache, intellisence, and other temporary files in Windows/Linux system folder. I will continue to manually locate such big files and delete them.

Thanks.
